gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/MaximumVersionComparatorTest.groovy

```diff
@@ -0,0 +1,68 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.dist
+
+import com.palantir.sls.versions.OrderableSlsVersion
+import com.palantir.sls.versions.SlsVersionMatcher
+import spock.lang.Specification
+
+class MaximumVersionComparatorTest extends Specification {
+    def comparesMatchers() {
+        given:
+        def m1 = MaximumVersion.valueOf("2.6.x")
+        def m2 = MaximumVersion.valueOf("2.x.x")
+        expect:
+        MaximumVersionComparator.INSTANCE.compare(m1, m2) < 0
+        MaximumVersionComparator.INSTANCE.compare(m2, m1) > 0
```

Comment:
I don't think this makes sense. `2.x.x` is neither larger nor smaller than `2.6.x`: it is simultaneously smaller, equal, and larger.